src/foam/nanos/crunch/SessionCrunchCache.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch',
+  name: 'SessionCrunchCache',
+
+  javaImports: [
+    'foam.core.X',
+    'foam.core.Detachable',
+    'foam.dao.AbstractSink',
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'java.util.ArrayList',
+    'java.util.Arrays',
+    'java.util.Map',
+    'java.util.List',
+    'java.util.concurrent.ConcurrentHashMap',
+    'static foam.mlang.MLang.*'
+  ],
+
+  properties: [
+    {
```

Comment:
Easier to just have a single property called seqId which is an AtomicLong and get rid of the two SeqId properties. When the ID is updated, set the two maps to null and synchronize access to thost collections with a wrapper that checks the sequence number. Then, the map doesn't need to be concurrent because it will be immutable after created so it will be faster.

Reply:
> and synchronize access to thost collections with a wrapper that checks the sequence number
can you explain this part differently?

Reply:
> immutable after created

So... fill the map with all the possible `capabilityId`s when cache is invalidated?

Reply:
Something like 

public synchronized checkCache(long seqId) {
  if ( getSeqId() != seqId ) {
     getPrerequisites().clear();
     getDependents().clear();
  }
}

public synchronized List<String> getPrerequsites(long seqId, String capId) {
  checkCache(seqId);
  if ( getPrerequisites().get(capId) == null ) {
     getPrerequisites().put(capId, createPrerequsitesList(capId));
  }
  return getPrerequisites().get(capId);
}

// same for getDependents(long seqId, String capId)

Reply:
I wonder how I didn't notice that method name conflict before. Updated now